content/resources/user/gen3-client.md

```diff
@@ -52,60 +53,105 @@ Now you should be able to run the tool on the command-line from any directory by
 
-GUIDs are generated on the back-end, not submitted by users, and they are stored in the property `object_id`. The GUID or `object_id` for a submitted data file can be obtained via graphQL query or viewing the data file json record in the graphical model of the project.
+GUIDs are generated on the back-end, not submitted by users, and they are stored in the property `object_id`. In the past, GUIDs must be inserted into the common before actual data files can be uploaded. However, a recent update to the Gen3 data commons allows user to submit data files without pre-generated GUIDs. Instead, a random GUID will be generated and assigned to each data file that has been submitted via the `gen3-client upload` command.
+
+Example Usage:
+```
+* For uploading a single file:
+gen3-client upload --profile=<profile_name> --upload-path=<path_to_files/data.bam>
+* For uploading all files within an folder:
+gen3-client upload --profile=<profile_name> --upload-path=<path_to_files/folder/>
+* Can also support regex such as:
+gen3-client upload --profile=<profile_name> --upload-path=<path_to_files/folder/*>
+* Or:
+gen3-client upload --profile=<profile_name> --upload-path=<path_to_files/*/folder/*.bam>
+
+gen3-client upload --profile=bob --file=test.gif
+
+Uploading data ...
+test.gif  3.64 MiB / 3.64 MiB [==========================================================================================] 100.00%
+Successfully uploaded file "test.gif" to GUID 65f5d77c-1b2a-4f41-a2c9-9daed5a59f14.
+1 files uploaded.
+```
+
+### Local submission history:
+In this mode, the application will keeps tracking which local file has already been submitted to avoid potential duplication in submissions. These information are kept in an JSON file under the same user folder as where the `config` file lives, for example:
+```
+Mac/Linux: /Users/Bob/.gen3/<your_config_name>_history.json
+Windows: C:\Users\Bob\.gen3\<your_config_name>_history.json
+```
 
-Once the GUIDs have been assigned to the data files for upload, the gen3-client can be used to upload files to object storage using the `gen3-client upload` command.
+Each object in the history JSON file is a key/value pair of the full file path of a file and GUID it associates with.
 
-Example:
+Example of a History JSON File:
+```
+{
+  "/Users/Bob/test.gif":"65f5d77c-1b2a-4f41-a2c9-9daed5a59f14"
+}
 ```
-gen3-client upload --profile <profile-name> --guid <GUID> --file=<filename>
 
-gen3-client upload --profile bob --guid dg.7519/b4642430-8c6e-465a-8e20-97c458439387 --file=test.gif
+
+* * *
+## 4) Upload a Single Data File Using a GUID
+* * *
+In this mode, a record that representing the data file with a pre-generated GUID must be already existing in the Gen3 data common's database of the `Indexd` service. The GUID or `object_id` for a submitted data file can then be obtained via graphQL query or viewing the data file JSON record in the graphical model of the project.
```

Comment:
"In this mode, a record that represents the data file with a pre-generated GUID must already exist..."